src/components/editor/Bindings/Backfill/index.tsx

```diff
@@ -0,0 +1,194 @@
+    const currentCollection = useResourceConfig_currentCollection();
+    const collections = useResourceConfig_collections();
+
+    const backfilledCollections = useResourceConfig_backfilledCollections();
+    const setBackfilledCollections =
+        useResourceConfig_setBackfilledCollections();
+
+    const backfillAllBindings = useResourceConfig_backfillAllBindings();
+    const allBindingsDisabled = useResourceConfig_allBindingsDisabled();
+
+    const selected = useMemo(() => {
+        if (bindingIndex === -1) {
+            return backfillAllBindings;
+        }
+
+        return currentCollection
+            ? backfilledCollections.includes(currentCollection)
+            : false;
+    }, [
+        backfillAllBindings,
+        backfilledCollections,
+        bindingIndex,
+        currentCollection,
+    ]);
+
+    const value: BooleanString = useMemo(
+        () => (selected ? 'true' : 'false'),
+        [selected]
+    );
+
+    const draftSpec = useMemo(
+        () =>
+            draftSpecs.length > 0 && draftSpecs[0].spec ? draftSpecs[0] : null,
+        [draftSpecs]
+    );
+
+    const evaluateServerDifferences = useCallback(
+        (increment: BooleanString) => {
+            if (bindingIndex === -1) {
+                return (
+                    (backfillAllBindings && increment === 'false') ||
+                    (!backfillAllBindings && increment === 'true')
+                );
+            }
+
+            if (currentCollection) {
+                return increment === 'true'
+                    ? !backfilledCollections.includes(currentCollection)
+                    : backfilledCollections.includes(currentCollection);
+            }
+
+            return false;
+        },
+        [
+            backfillAllBindings,
+            backfilledCollections,
+            bindingIndex,
+            currentCollection,
+        ]
+    );
+
+    const handleClick = useCallback(
+        (increment: BooleanString) => {
+            const serverUpdateRequired = evaluateServerDifferences(increment);
+
+            if (draftSpec && serverUpdateRequired) {
+                setFormState({ status: FormStatus.UPDATING });
+
+                const singleBindingUpdate =
+                    bindingIndex > -1 && currentCollection;
+
+                const bindingMetadata: BindingMetadata[] = singleBindingUpdate
+                    ? [{ collection: currentCollection, bindingIndex }]
+                    : [];
+
+                updateBackfillCounter(
+                    draftSpec,
+                    increment,
+                    bindingMetadata
+                ).then(
+                    () => {
+                        const targetCollection = singleBindingUpdate
+                            ? currentCollection
+                            : undefined;
+
+                        setBackfilledCollections(increment, targetCollection);
+                        setFormState({ status: FormStatus.UPDATED });
```

Comment:
This seems to not be resetting the state properly. I think it needs to pass an empty `error` maybe? 

Once you get a network error if you then click again and it works the error just hangs around. We should probably reset the state of failed/error when they click?
![image](https://github.com/estuary/ui/assets/270078/7e81ac1a-cb50-4430-bd6b-902a37153216)

Reply:
Playing with this for a second... I think we should probably reset the error up above to make the UX a bit more clear if another error happens.

Reply:
Not sure why I assumed that the form state store cleared errors when transitioning from an idle to an active state.